code/modules/cm_preds/yaut_weapons.dm

```diff
@@ -403,24 +403,49 @@
 	if(target.stat != DEAD)
 		return ..()
 
-	if(!ishuman(target))
-		to_chat(user, SPAN_WARNING("You can only use this dagger to flay humanoids!"))
+	if(!iscarbon(target))
+		to_chat(user, SPAN_WARNING("This creature is not worthy!"))
 		return
 
-	var/mob/living/carbon/human/victim = target
-
 	if(!HAS_TRAIT(user, TRAIT_SUPER_STRONG))
 		to_chat(user, SPAN_WARNING("You're not strong enough to rip an entire humanoid apart. Also, that's kind of fucked up.")) //look at this dumbass
 		return
 
-	if(issamespecies(user, victim))
-		to_chat(user, SPAN_HIGHDANGER("ARE YOU OUT OF YOUR MIND!?"))
+	if(isxeno(target))
+		var/mob/living/carbon/xenomorph/xeno = target
+		if(do_after(user, 1 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, xeno))
+			to_chat(user, SPAN_WARNING("You start cutting [xeno] appart."))
+			playsound(loc, 'sound/weapons/pierce.ogg', 25)
+			if(do_after(user, 4 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, xeno))
+				to_chat(user, SPAN_WARNING("You pierce [xeno] exoeskeleton with precise cuts"))
+				create_leftovers(xeno, has_meat = FALSE, skin_amount = TRUE)
+				playsound(loc, 'sound/weapons/slashmiss.ogg', 25)
+				if(do_after(user, 4 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, xeno))
+					to_chat(user, SPAN_WARNING("You harvest [xeno] insides"))
+					create_leftovers(xeno, has_meat = TRUE, skin_amount = FALSE,)
+					create_leftovers(xeno, has_meat = TRUE, skin_amount = FALSE,)
+					playsound(loc, 'sound/weapons/slash.ogg', 25)
+					if(do_after(user, 4 SECONDS, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, xeno))
+						to_chat(user, SPAN_WARNING("You extract [xeno] meat leaving behind a pile of goo"))
+						var/obj/item/xeno_tail/tail = new /obj/item/xeno_tail(get_turf(xeno)) // the tail remains
+						tail.name = "[xeno.name] tail"
+						playsound(loc, 'sound/bullets/bullet_impact3.ogg', 25)
+						xeno.gib() //we dont need you anymore
+						return
```

Comment:
convert this into a separate proc, same with the human

`if(isxeno(target)`
`  xeno_flay(target, user)`
`else if(ishuman(target)`
`  human_flay(target, user)`
`else return`

Reply:
thanks for reviewing my shitcode 😿